### PR TITLE
[PyTorch] Set net.eval and net.train explicitly

### PR DESCRIPTION
**Pull request by AnirudhDagar**
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by AnirudhDagar** [comment]
Addresses #1093 as well.

**Comment by mli** [comment]
Job d2l-en/PR-1110/3 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1110/

**Comment by AnirudhDagar** [comment]
@StevenJokes Yes, I know. But once you switch to net.eval(), you need to switch back to net.train() in the training loop.

**Comment by AnirudhDagar** [comment]
@astonzhang Can you PTAL?

**Comment by mli** [comment]
Job d2l-en/PR-1110/4 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1110/

**Comment by AnirudhDagar** [comment]
@mli @astonzhang PTAL

**Comment by goldmermaid** [comment]
LGTM!

File: chapter_convolutional-neural-networks/lenet.md [hunk 1]
```diff
@@ -244,6 +244,7 @@ def evaluate_accuracy_gpu(net, data_iter, ctx=None):  #@save
 ```{.python .input}
 #@tab pytorch
 def evaluate_accuracy_gpu(net, data_iter, device=None): #@save
+    net.eval() # Set the model to evaluation mode
```
Comment (astonzhang) [inline chapter_convolutional-neural-networks/lenet.md:247]: 2 space before inline comments

File: chapter_linear-networks/softmax-regression-scratch.md [hunk 2]
```diff
@@ -355,9 +355,21 @@ Similarly, we can evaluate the accuracy for any model `net` on a dataset
...
+        metric.add(accuracy(net(X), y), sum(y.shape))
+    return metric[0] / metric[1]
+```
+
+```{.python .input}
+#@tab pytorch
 def evaluate_accuracy(net, data_iter):  #@save
     """Compute the accuracy for a model on a dataset."""
+    if isinstance(net, torch.nn.Module):
+        net.eval() # Set the model to evaluation mode
```
Comment (astonzhang) [inline chapter_linear-networks/softmax-regression-scratch.md:372]: 2 space before inline comments

File: chapter_linear-networks/softmax-regression-scratch.md [hunk 3]
```diff
@@ -538,12 +550,30 @@ We will leverage the `Animator` class to visualize
...
+
+```{.python .input}
+#@tab pytorch
 def train_ch3(net, train_iter, test_iter, loss, num_epochs, updater):  #@save
     """Train a model (defined in Chapter 3)."""
     animator = Animator(xlabel='epoch', xlim=[1, num_epochs], ylim=[0.3, 0.9],
                         legend=['train loss', 'train acc', 'test acc'])
     for epoch in range(num_epochs):
+        if isinstance(net, torch.nn.Module):
+            net.train() # Set the model to training mode
```
Comment (astonzhang) [inline chapter_linear-networks/softmax-regression-scratch.md:576]: 2 space before inline comments

**Comment by mli** [comment]
Job d2l-en/PR-1110/5 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1110/

**Comment by mli** [comment]
Job d2l-en/PR-1110/6 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1110/

**Comment by mli** [comment]
Job d2l-en/PR-1110/7 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1110/

**Comment by AnirudhDagar** [comment]
@astonzhang @mli PTAL again!
These changes can break the saved chapter train functions if something is wrong. Hence want this to be carefully reviewed and be completely correct before being merged.

Thanks

**Comment by astonzhang** [comment]
@alexanderchr  Does this PR address your issue? Thanks.

**Comment by alexanderchr** [comment]
@astonzhang It does, thanks!

**Comment by astonzhang** [comment]
@AnirudhDagar I've merged it. Can you double-check the experimental results that depend on this modification and make sure they still look good?
